# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pkg_patch/pkg_libpcap.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -ruN libpcap-0.9.2.orig/pcap-linux.c libpcap-0.9.2/pcap-linux.c
--- libpcap-0.9.2.orig/pcap-linux.c	2005-07-07 04:04:34.000000000 +0200
+++ libpcap-0.9.2/pcap-linux.c	2005-07-13 19:35:22.000000000 +0200
@@ -149,7 +149,7 @@
 #include <linux/filter.h>
 #endif
 
-#ifndef __GLIBC__
+#if !(defined(__GLIBC__) || defined(__dietlibc__))
 typedef int		socklen_t;
 #endif
 
